src/data/common/Dhis2DataFormRepository.ts

```
@@ -292,13 +291,21 @@ export class Dhis2DataFormRepository implements DataFormRepository {
 
         return {
             type: ruleType,
-            dataElements: rule.dataElements.map(dataElementCode => dataElements[dataElementCode]?.id ?? ""),
+            dataElements: this.getTotalDataElements(rule, dataElements),
             relatedDataElements: relatedDataElements,
-            condition: rule.condition,
+            conditions:
+                rule.type === "option" ? rule.conditions.map(condition => condition.condition) : [rule.condition],
             formula: formula,
         };
     }
 
+    private getTotalDataElements(rule: ConditionRule, dataElements: Record<string, DataElement>) {
+        const extractedDataElements =
+            rule.type === "option" ? rule.conditions.flatMap(condition => condition.dataElements) : rule.dataElements;
+
+        return extractedDataElements.map(dataElementCode => dataElements[dataElementCode]?.id ?? "");
```

Since we're getting the id's here, can we rename the name to `getTotalDataElementIds`? And including the type should be useful here: 

```private getTotalDataElementIds(rule: ConditionRule, dataElements: Record<string, DataElement>): Id[]```

Is safe for this method to return an empty string?  I think we can either throw an error or at least log a warning to console and ignore empty strings.